rust-runtime/aws-smithy-runtime/src/client/retries/strategy/standard.rs

```diff
@@ -51,14 +51,28 @@ impl StandardRetryStrategy {
         Default::default()
     }
 
-    fn release_retry_permit(&self) -> ReleaseResult {
+    fn release_retry_permit(&self, token_bucket: &TokenBucket) -> ReleaseResult {
         let mut retry_permit = self.retry_permit.lock().unwrap();
         match retry_permit.take() {
             Some(p) => {
-                drop(p);
+                // Retry succeeded: reward success and forget permit if configured, otherwise release permit back
```

Comment:
I think this logic made more sense outside of `release_retry_permit`. Adding it here overloads the function's responsibility.